### PR TITLE
Fix negated messages and improve certain patterns

### PR DESCRIPTION
**Pull request by streichsbaer**
Hi @ajinabraham , 

I've been going through mobsfscan and noticed some inconsistencies with messages related to best-practices.
I've also improved 2 patterns to cover more use-cases. 

Please check whether my updates make sense.

Thanks

### CODE REVIEW

File: mobsfscan/rules/patterns/ios/swift/best_practices.yaml [hunk 1]
```diff
@@ -68,9 +68,13 @@
...
+  message: This app may disable Keyboard cache. It must be disabled for all sensitive data inputs.
   input_case: exact
-  pattern: \.autocorrectionType = \.no
+  pattern:
+    - \.autocorrectionType\s*=\s*\.no
+    - \.autocorrectionType\s*=\s*\.No
+    - UITextAutocorrectionTypeNo
+    - textField\.disableAutocorrection\s*\(\s*true\s*\)
   severity: INFO
   type: Regex
```
Comment (ajinabraham) [inline mobsfscan/rules/patterns/ios/swift/best_practices.yaml:79]: This should be `RegexOr`

File: mobsfscan/rules/patterns/ios/swift/best_practices.yaml [hunk 2]
```diff
@@ -68,9 +68,13 @@
...
 - id: ios_keyboard_cache
-  message: This app does not disable Keyboard cache. It must be disabled for all sensitive data inputs.
+  message: This app may disable Keyboard cache. It must be disabled for all sensitive data inputs.
   input_case: exact
-  pattern: \.autocorrectionType = \.no
+  pattern:
+    - \.autocorrectionType\s*=\s*\.no
+    - \.autocorrectionType\s*=\s*\.No
+    - UITextAutocorrectionTypeNo
+    - textField\.disableAutocorrection\s*\(\s*true\s*\)
```
Comment (ajinabraham) [inline mobsfscan/rules/patterns/ios/swift/best_practices.yaml:77]: Can `textField` be omitted?
`disableautocorrection` is an instance method https://developer.apple.com/documentation/swiftui/text/disableautocorrection(_:)

File: mobsfscan/rules/patterns/ios/swift/swift_rules.yaml [hunk 3]
```diff
@@ -70,9 +70,12 @@
 - id: ios_biometric_bool
   message: Biometric authentication should be based on Keychain, not based on bool.
   input_case: exact
-  pattern: \.evaluatePolicy\(.deviceOwnerAuthentication
+  pattern:
+    - \.evaluatePolicy\(.deviceOwnerAuthentication
```
Comment (ajinabraham) [inline mobsfscan/rules/patterns/ios/swift/swift_rules.yaml:74]: Can these two be replaced with  `\.evaluatePolicy\(.*\.deviceOwnerAuthentication` ?

File: mobsfscan/rules/patterns/ios/swift/swift_rules.yaml [hunk 4]
```diff
@@ -70,9 +70,12 @@
...
+    - \.evaluatePolicy\(LAPolicy.deviceOwnerAuthentication
+    - evaluatePolicy:LAPolicyDeviceOwnerAuthentication
```
Comment (ajinabraham) [inline mobsfscan/rules/patterns/ios/swift/swift_rules.yaml:76]: This one (`LAPolicyDeviceOwnerAuthentication`)  looks like Objective C code?

https://developer.apple.com/documentation/localauthentication/lapolicy/lapolicydeviceownerauthentication?language=objc

Should this be moved to ` mobsfscan/rules/patterns/ios/objectivec/best_practices.yaml ` ?

File: mobsfscan/rules/patterns/ios/swift/swift_rules.yaml [hunk 5]
```diff
@@ -70,9 +70,12 @@
...
   severity: WARNING
-  type: Regex
+  type: RegexOr
```
Comment (ajinabraham) [inline mobsfscan/rules/patterns/ios/swift/swift_rules.yaml:78]: Revert if the `LAPolicyDeviceOwnerAuthentication` rule is moved to Objective C

**Review by ajinabraham — CHANGES_REQUESTED** [review]
Hey @streichsbaer 

Thanks for the PR.

 I've been going through mobsfscan and noticed some inconsistencies with messages related to best-practices.

> I think you got confused here. These best practices rules are special case and their descriptions are negated by design. Unlike other rules which are triggered when we find those patterns in code, these are triggered only when we do not find a pattern after scanning the entire input files. See: https://github.com/MobSF/mobsfscan/blob/5fa3a77d1726f654c93794a1172ef14d077e03ad/mobsfscan/mobsfscan.py#L112-L135
Hence the description says, we `do not` find x best practise after scanning the entire code.

I've also improved 2 patterns to cover more use-cases.

> This is really useful, can you update the PR to remove the negation fix and include additional patterns?

File: mobsfscan/rules/patterns/ios/swift/swift_rules.yaml (hunk 3 above)
Comment (streichsbaer) [reply mobsfscan/rules/patterns/ios/swift/swift_rules.yaml:74]: yes, that's cleaner, will update shortly.

File: mobsfscan/rules/patterns/ios/swift/swift_rules.yaml (hunk 4 above)
Comment (streichsbaer) [reply mobsfscan/rules/patterns/ios/swift/swift_rules.yaml:76]: Yes, will take care of it.

Comment (streichsbaer) [reply mobsfscan/rules/patterns/ios/swift/swift_rules.yaml:76]: Have added it to `objectivec/objective_c_rules.yaml` to be consistent as the swift one was not under best practices either.

File: mobsfscan/rules/patterns/ios/swift/best_practices.yaml [hunk 6]
```diff
@@ -80,7 +84,7 @@
     reference: https://github.com/MobSF/owasp-mstg/blob/master/Document/0x06d-Testing-Data-Storage.md#finding-sensitive-data-in-the-keyboard-cache-mstg-storage-5
 - id: ios_uiwebview_safari_view
   message: >-
-    This app uses SFSafariViewController or UIWebView. It 
+    This app uses SFSafariViewController or UIWebView. It
```
Comment (streichsbaer) [inline mobsfscan/rules/patterns/ios/swift/best_practices.yaml:87]: Then isn't this is incorrectly positive as a best_practice?

File: mobsfscan/rules/patterns/ios/swift/swift_rules.yaml (hunk 4 above)
Comment (ajinabraham) [reply mobsfscan/rules/patterns/ios/swift/swift_rules.yaml]: Yes, you are right.

File: mobsfscan/rules/patterns/ios/swift/best_practices.yaml (hunk 6 above)
Comment (ajinabraham) [reply mobsfscan/rules/patterns/ios/swift/best_practices.yaml:87]: Yeah, you are right. I will move this shortly.

**Comment by ajinabraham** [comment]
Did some QA in `main` branch and resolved merge conflict, now the only remaining this is to revert  to `do not` in rule description.

**Comment by streichsbaer** [comment]
Alright, changes have been made. Feel free to merge it or let me know if anything else needs adjusting.

**Review by ajinabraham — APPROVED** [review]
LGTM 👍

**Pull request merged**